src/composables/useUiDesign.ts

```diff
@@ -0,0 +1,100 @@
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Affero General Public License and its applicable Additional Terms for
+ * LinID Identity Manager along with this program. If not, see <http://www.gnu.org/licenses/> for the GNU Affero
+ * General Public License version 3 and <http://www.linagora.com/licenses/> for the Additional Terms applicable to the
+ * LinID Identity Manager software.
+ */
+
+import type { ComputedRef } from 'vue';
+import { computed } from 'vue';
+import { getUiDesign } from '../services/uiDesignService';
+import type {
+  UiDesign,
+  UiDesignNamespace,
+  UiDesignValue,
+} from '../types/uiDesign';
+
+/**
+ * Composable providing access to the shared UI design configuration.
+ *
+ * Allows retrieving UI configuration values for a given namespace and key,
+ * with automatic fallback to the `default` namespace if the value is not defined.
+ * @returns An object containing the `ui` function.
+ */
+export function useUiDesign() {
+  const cfg: ComputedRef<UiDesign> = computed(() => getUiDesign());
+
+  /**
+   * Retrieves a nested value from an object using dot-separated keys.
+   * @param obj - The object to traverse.
+   * @param path - The path string, e.g., 'namespace.key.subkey'.
+   * @returns The value at the given path, or undefined if not found.
+   */
+  function getNested(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    obj: Record<string, any>,
+    path: string
+  ): UiDesignValue | UiDesignNamespace | undefined {
+    return path
+      .split('.')
+      .reduce<UiDesignValue | UiDesignNamespace | undefined>((acc, key) => {
+        if (acc && typeof acc === 'object') {
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          return (acc as Record<string, any>)[key];
+        }
+        return undefined;
+      }, obj);
+  }
+
+  /**
+   * Retrieve a UI configuration value from a namespace with fallback to `default`.
+   * Always returns a UiDesignValue. Throws if the value is not found or is a nested object.
+   * @param namespace - The namespace to look for (can be nested using dot notation).
+   * @param type - The key within the namespace (can be nested using dot notation).
+   * @returns The primitive configuration value.
+   */
+  function ui(namespace: string, type: string): UiDesignValue {
+    let value = getNested(cfg.value, `${namespace}.${type}`);
+
+    // fallback to default
+    if (value === undefined) {
+      value = getNested(cfg.value, `default.${type}`);
+    }
+
+    if (value === undefined) {
+      throw new Error(
+        `[UiDesign] Value not found for '${namespace}.${type}' and no default fallback.`
+      );
+    }
+
+    // ensure we only return a primitive
+    if (typeof value === 'object') {
```

Comment:
If value is null the condition `typeof value === 'object'` is true.
```suggestion
    if (typeof value === 'object' && value !== null) {
```